[DEVOPS-1181] Validating Lighthouse scores against configured pass values

g-foster2020 opened the pull request.

Commit:
Validating Lighthouse scores against configured pass values

Commit:
Validating Lighthouse scores against configured pass values

Commit:
Validating Lighthouse scores against configured pass values

kierenevans commented on src/_base/docker/image/lighthouse/root/app/run.sh.twig:
```diff
@@ -0,0 +1,67 @@
+           --chrome-flags="--headless --no-sandbox=true" \
+           --output json \
+           --output-path=/home/headless/lighthouse-results.json \
+           ${TARGET_URL}
```
Fix from shellcheck:
```suggestion
           "${TARGET_URL}"
```

kierenevans commented on src/_base/harness/attributes/common.yml:
```diff
@@ -156,6 +156,22 @@ attributes.default:
+        score: 0
+      performance:
+        enabled: 0
+        score: 0
```
For consistency with other true/false values:
```suggestion
    success-thresholds:
      pwa:
        enabled: false
        score: 0
      seo:
        enabled: false
        score: 0
      best-practices:
        enabled: false
        score: 0
      accessibility:
        enabled: false
        score: 0
      performance:
        enabled: false
        score: 0
```

kierenevans commented on src/_base/docker/image/lighthouse/Dockerfile.twig:
```diff
@@ -3,7 +3,10 @@ FROM yukinying/chrome-headless-browser:latest
 USER root
 
 RUN apt update
+RUN apt install -y jq
+RUN apt install -y bc
 RUN apt install -y npm
```
The image doesn't need to be the smallest in final size or layers due to not being pushed anywhere, but it would still be a small speed improvement to work out the dependencies for all three packages in one go:
```suggestion
RUN apt update \
 && apt install -y bc jq npm
```

kierenevans commented on src/_base/harness/config/confd.yml:
```diff
@@ -18,6 +18,7 @@ confd('harness:/'):
   - { src: docker/image/console/root/usr/local/etc/php/conf.d/docker-php-ext-tideways.ini }
   - { src: docker/image/console/root/usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini }
   - { src: docker/image/lighthouse/Dockerfile }
+  - { src: docker/image/lighthouse/root/app/run.sh }
```
This will need replicating into the `src/akeneo/harness/config/confd.yml` and the rest of the harness `confd.yml`s.

kierenevans reviewed (dismissed):
The confd.yml comment and shellcheck fix are needed and the rest are optional.

Could we have a Jenkinsfile.twig addition to run these tests by default, even if it wouldn't fail the build? There's one in _base and one in drupal8 so would need to be in both.

I'd also like to see this tested in this repository's `./test` script like we do for the other Jenkins tests: https://github.com/inviqa/harness-base-php/blob/1.1.x/test#L88
This will allow us to hopefully not break project's use of the lighthouse testing.

g-foster2020 commented:
> The confd.yml comment and shellcheck fix are needed and the rest are optional.
> 
> Could we have a Jenkinsfile.twig addition to run these tests by default, even if it wouldn't fail the build? There's one in _base and one in drupal8 so would need to be in both.
> 
> I'd also like to see this tested in this repository's `./test` script like we do for the other Jenkins tests: https://github.com/inviqa/harness-base-php/blob/1.1.x/test#L88
> This will allow us to hopefully not break project's use of the lighthouse testing.

Great feedback as ever thanks. I will try to remember shellcheck from now on and the confd overrides.

I will implement all of these and your above suggestions.

g-foster2020 committed:
Update src/_base/docker/image/lighthouse/root/app/run.sh.twig

Co-authored-by: Kieren Evans <kierenevans@users.noreply.github.com>

g-foster2020 committed:
Update src/_base/harness/attributes/common.yml

Co-authored-by: Kieren Evans <kierenevans@users.noreply.github.com>

g-foster2020 committed:
Update src/_base/docker/image/lighthouse/Dockerfile.twig

Co-authored-by: Kieren Evans <kierenevans@users.noreply.github.com>

Commit:
Suggested changes to improve lighthouse integration into harness

Commit:
Merge branch 'feature/DEVOPS-1181-lighthouse-additional' of https://github.com/inviqa/harness-base-php into feature/DEVOPS-1181-lighthouse-additional

g-foster2020 committed:
[DEVOPS-1181] lighthouse local (#559)

* DEVOPS-1181 Enable lighthouse globally, but exit immediately in local dev

* Allow exit code zero through container check

* Fixes after testing

* Run lighthouse as part of the same compose project name

* Ignore certificate errors

Co-authored-by: Kieren Evans <kevans@inviqa.com>

Commit:
Try to make the image build quicker

Commit:
Try to make the image build quicker

Commit:
Try to make the image build quicker

Commit:
Try to make the image build quicker

Commit:
Try to make the image build quicker

Commit:
Try to make the image build quicker

Commit:
Try to make the image build quicker

Commit:
Try to make the image build quicker

Commit:
Try to make the image build quicker

Commit:
Try to make the image build quicker

Commit:
Try to make the image build quicker

Commit:
Try to make the image build quicker

Commit:
Try to make the image build quicker

Commit:
Try to make the image build quicker

Commit:
Extra info to help when there is an issue

Commit:
Extra info to help when there is an issue

Commit:
Working Spryker subdomain for default lighthouse test

Commit:
Working Spryker subdomain for default lighthouse test

Commit:
Working Spryker subdomain for default lighthouse test

Commit:
Working Spryker subdomain for default lighthouse test

Commit:
Temporary change to enable debugging of Spryker lighthouse test

Commit:
Working Spryker subdomain for default lighthouse test

Commit:
Working Spryker subdomain for default lighthouse test

Commit:
Working Spryker subdomain for default lighthouse test

Commit:
Working Spryker subdomain for default lighthouse test

Commit:
Wait for Spryker to start before running lighthouse test

Commit:
Wait for Spryker to start before running lighthouse test

Commit:
Remove wait for Spryker to start before running lighthouse test

Commit:
Remove unneeded override file

kierenevans committed:
Fix spryker URLs not being aliased correctly in static mode

Commit:
Working Spryker subdomain for default lighthouse test

kierenevans committed:
Merge remote-tracking branch 'origin/1.1.x' into feature/DEVOPS-1181-lighthouse-additional

kierenevans committed:
DEVOPS-1181 Load up NVM

kierenevans committed:
Use data path for Spryker twig cache

Despite being deprecated in the 202009.0 release, we don't want to make src/Generated be web-writable

kierenevans committed:
Merge branch 'feature/use-data-path-for-spryker-twig-cache' into feature/DEVOPS-1181-lighthouse-additional

kierenevans committed:
DEVOPS-1181 Switch back to Yves testing as it's customer facing

kierenevans committed:
Revert "Temporary change to enable debugging of Spryker lighthouse test"

This reverts commit 3598f56338c3788cac58b9243817346b36eeef0c.

kierenevans committed:
nvm init handles path. Already at v15.11.0, so PATH doesn't take effect
